session/session_test.go

```diff
@@ -34,6 +37,55 @@ func (s *SessionTestSuite) TestAll() {
 	s.Equal("value2", all["key2"])
 }
 
+func (s *SessionTestSuite) TestExists() {
+	s.session.Put("foo", "bar")
+	s.True(s.session.Exists("foo"))
+	s.session.Put("baz", nil)
+	s.False(s.session.Has("baz"))
+	s.True(s.session.Exists("baz"))
+	s.False(s.session.Exists("bogus"))
+	s.True(s.session.Exists("foo"))
+}
+
+func (s *SessionTestSuite) TestFlash() {
+	s.session.Flash("foo", "bar").
+		Flash("bar", 0).
+		Flash("baz", true)
+
+	s.True(s.session.Has("foo"))
+	s.Equal("bar", s.session.Get("foo"))
+	s.Equal(0, s.session.Get("bar"))
+	s.Equal(true, s.session.Get("baz"))
+
+	s.session.ageFlashData()
+
+	s.True(s.session.Has("foo"))
+	s.Equal("bar", s.session.Get("foo"))
+	s.Equal(0, s.session.Get("bar"))
+
+	s.session.ageFlashData()
+
+	s.False(s.session.Has("foo"))
```

Comment:
Why it's `true` in line 62, but it's `false` in line 68? I think they should both `false`, because `foo` is got in lin 56.

Reply:
In the first `ageFlashData` call, it removes all keys that exist in `_flash.old` and adds all new flash keys into it. However, when we call it a second time, `"foo"` and `"bar"` are already in `_flash.old`, so they will be removed just like they were in the first call (similar to the `http` request). [This function will only be called once per request.]